plugins/inputs/http_response/http_response.go

```diff
@@ -130,15 +131,24 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 	// Start Timer
 	start := time.Now()
 	resp, err := h.client.Do(request)
+
 	if err != nil {
+		fields["http_response_code"] = -1
+		fields["response_string_match"] = 0
+		if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+			fields["response_timeout"] = time.Since(start).Seconds()
+		} else {
+			fields["response_time"] = time.Since(start).Seconds()
+		}
+		fields["connection_failed"] = 1
 		if h.FollowRedirects {
-			return nil, err
+			return fields, err
```

Comment:
In general, returning an error indicates that the value is not valid, if we are returning fields then we shouldn't have an error.  Later on, we will need to make sure we don't call AddFields if it is nil.